src/rcc.rs

```diff
@@ -121,12 +123,16 @@ pub struct AHB {
 }
 
 impl AHB {
-    pub(crate) fn enr(&mut self) -> &rcc::AHBENR {
+    pub(crate) fn new() -> Self {
```

Comment:
This now allows creating a reference to the instance out of thin air. I'd like to make this function unsafe, because this theoretically allows access to the register from multiple contexts.